build/yamls/flow-aggregator.yml

```diff
@@ -131,11 +131,20 @@ data:
     # If no L4 transport proto is given, we consider tcp as default.
     #externalFlowCollectorAddr: ""
 
-    # Provide flow export interval as a duration string. This determines how often the flow aggregator exports flow
-    # records to the flow collector.
-    # Flow export interval should be greater than or equal to 1s (one second).
-    # Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
-    #flowExportInterval: 60s
+    # Provide the active flow record timeout as a duration string. This determines
+    # how often the flow aggregator exports the active flow records to the flow
+    # collector. Thus, for flows with a continuous stream of packets, a flow record
+    # will be exported to the collector once the elapsed time since the last export
+    # event is equal to the value of this timeout.
+    # Defaults to "60s". Follow the time units of duration type.
+    #activeFlowRecordTimeout: 60s
+
+    # Provide the inactive flow record timeout as a duration string. This determines
+    # how often the flow aggregator exports the inactive flow records to the flow
+    # collector. A flow record is considered to be inactive if no matching record
+    # has been received from any Node in the cluster since the last export event.
+    # Defaults to "120s". Follow the time units of duration type.
+    #inactiveFlowRecordTimeout: 120s
```

Comment:
I'm trying to understand the distinction between "inactive" timeout for the FlowAggregator and "idle" timeout for the FlowExporter. In particular I am confused by the fact that `inactiveFlowRecordTimeout` is larger than `activeFlowRecordTimeout`, while in the FlowExporter the idle timeout is (as expected) smaller than the active timeout. Will an inactive timeout ever be triggered in that case? With the default values, it seems to be that if we stop receiving records from the agents, after 60s the active timeout will be triggered, even though one would expect the inactive timeout to be the effective one in that case.

Reply:
I tried to make a distinction between "inactive" timeout at Flow Aggregator and "idle" timeout at flow exporter. With idle timeout at Flow Exporter, we do not delete the flow record in the cache, rather wait for it to become active or time out in conntrack before deleting it. In the aggregation process, following the [RFC doc](https://tools.ietf.org/html/rfc6183#section-5.3.1), we can have a policy to delete the record from the cache once it becomes inactive, i.e., no records received from a certain amount of time.

I went with a higher value for `inactiveFlowRecordTimeout` compared to `activeFlowRecordTimeout`, to cover the cases of idle flow records, which could get paused for some time before becoming active again at Flow Exporter at Antrea agent.
In these cases, we can take a different route, where we delete the flow records at the aggregation process and create the flow record again when the flow records are received for "idle flows" at the flow exporter. We are not building any state for the flow record, so we can delete and recreate the record again.

I agree with your argument that it seems odd that we will be sending records with active expiry for inactive aggregate flow records before sending the record because of inactive expiry. As described above, it's better to go with a different route for idle record cases by keeping the `inactiveFlowRecordTimeout` smaller than the `inactiveFlowRecordTimeout`. 

What values do you think are appropriate? `inactiveFlowRecordTimeout` should be more than the `activeFlowRecordTimeout` at the exporter, whose default value is 60s.